applications/crossbar/src/api_util.erl

```diff
@@ -868,17 +868,24 @@ ensure_content_type(CT) -> CT.
 %% (the final module in the chain) accepts this verb parameter pair.
 %% @end
 %%--------------------------------------------------------------------
--spec does_resource_exist(cb_context:context()) -> boolean().
--spec does_resource_exist(cb_context:context(), list()) -> boolean().
+-spec does_resource_exist(cb_context:context()) -> {boolean(), cb_context:context()}.
+-spec does_resource_exist(cb_context:context(), list()) -> {boolean(), cb_context:context()}.
 does_resource_exist(Context) ->
     does_resource_exist(Context, cb_context:req_nouns(Context)).
 
 does_resource_exist(Context, [{Mod, Params}|_]) ->
     Event = api_util:create_event_name(Context, <<"resource_exists.", Mod/binary>>),
-    Responses = crossbar_bindings:map(Event, Params),
-    crossbar_bindings:any(Responses) and 'true';
-does_resource_exist(_Context, _ReqNouns) ->
-    'false'.
+    Responses = crossbar_bindings:map(Event, [Context, Params]),
```

Comment:
Yes, to match how the rest of the callbacks operate. Having verbs on the URI isn't good practice (so `/action` shouldn't be there) anyway; should be `/things/{thing_id}/{sub_thing_or_id}` if this is accessing parts of `{thing_id}`. So `resource_exists(Context)` checks that the collection exists, `resource_exists(Context, ThingId)` checks that ThingId exists, etc... If you are acting on ThingId, should be done in the request params, imo.